docs/siem/reference/timeline-schema.asciidoc

```diff
@@ -0,0 +1,270 @@
+|`sort` |sort a|Object indicating how rows are sorted in the timeline's grid:
+
+* `columnId` (string): The ID of the column used to sort results.
+* `sortDirection` (string): The sort direction, which can be either `desc` or
+`asc`.
+
+|`templateTimelineId` |String a| A unique ID for Timeline templates. For
+timelines, the value is `null`.
+|`templateTimelineVersion` |Integer |Timeline template version number. For
+timelines, the value is `null`.
+// When creating timeline template via import, can just specify it to 1.
+// We use this version to avoid template timeline to be overwrite when updating
+// via import.
+// We take every positive int given from user as long as it is grater than
+// current value.
+|`timelineType` |String a|Indicates whether the timeline is a template or not,
+where:
+
+* `default`: Indicates a timeline used to actively investigate events.
+* `template`: Indicates a timeline template used with detection rules for
+displaying alerts in Timeline.
+
+|[[timeline-object-title]]`title` |String |The timeline's title.
+|`updated` |Float |The last time the timeline was updated, using a
+13-digit Epoch timestamp.
+|`updatedBy` |String |The user who last updated the timeline.
+|`version` |String |The timeline's version.
+|==============================================
+
+[[col-obj]]
+[discrete]
+==== columns object
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description
+
+|`aggregatable` |Boolean |Indicates whether the column field header can be
+aggregated on all indices.???
+// SIEM default indices
+|`category` |String |The ECS field set to which the column field header belongs.
+|`description` |String |UI column field description tooltip.
+|`example` |String |UI column field example tooltip.
+|`indexes` |String |List of all indices in which the column field header has
+the same type. `null` when all indices have the same type for the field.???
+// SIEM default indices
+|`id` |String |ECS field alias or user defined field name for the column header.???
+// As displayed in the UI
+|`name` |String |Reserved for future use, typically `null`.???
+// Can `name` be removed?
+|`searchable` |Boolean |Indicates whether the column field header is indexed
+for search on all indices.???
+// SIEM indices? When wouldn't the field be searchable?
+|`type` |String |Column field header type.
+|==============================================
+
+[[dataProvider-obj]]
+[discrete]
+==== dataProviders object
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description
+
+|`and` |dataProviders[] |Array containing dropzone query clauses using `AND`
+logic.
+|`enabled` |Boolean |Indicates if the dropzone query clause is enabled.
+|`excluded` |Boolean |Indicates if the dropzone query clause uses `NOT` logic.
+|`id` |String |The dropzone query clause's unique ID.
+|`kqlQuery` |String |Always an empty string (not used).???
+// Can be removed?
```

Comment:
I think we are following some reusable schema across Kibana, therefore we can sometimes find the property is there but we are not actually using it.